rotkehlchen/tests/api/test_ethereum_transactions.py

```diff
@@ -1552,3 +1563,132 @@ def test_force_redecode_evm_transactions(rotkehlchen_api_server: 'APIServer') ->
+                ), EvmEvent(
+                    tx_hash=make_evm_tx_hash(),
+                    sequence_index=0,
+                    timestamp=gnosis_pay_ts_2,
+                    location=Location.GNOSIS,
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.NONE,
+                    asset=A_EUR,
+                    amount=ONE,
+                    counterparty=CPT_GNOSIS_PAY,
+                ), EvmEvent(
+                    tx_hash=monerium_tx,
+                    sequence_index=0,
+                    timestamp=TimestampMS(1620000000),
+                    location=Location.ARBITRUM_ONE,
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.NONE,
+                    asset=A_ETH,
+                    amount=ONE,
+                    counterparty=CPT_MONERIUM,
+                ), EvmEvent(
+                    tx_hash=make_evm_tx_hash(),
+                    sequence_index=0,
+                    timestamp=TimestampMS(1630000000),
+                    location=Location.GNOSIS,
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.NONE,
+                    asset=A_ETH,
+                    amount=ONE,
+                )],
+            )
+
+    with (
+        patch('rotkehlchen.externalapis.monerium.Monerium.get_and_process_orders') as mock_monerium,  # noqa: E501
+        patch('rotkehlchen.externalapis.gnosispay.GnosisPay.maybe_update_event_with_api_data') as mock_gnosis_pay_update_events,  # noqa: E501
+        patch('rotkehlchen.externalapis.gnosispay.GnosisPay.query_remote_for_tx_and_update_events') as mock_gnosis_pay_query_remote,  # noqa: E501
+        patch.object(rotki.chains_aggregator.gnosis.transactions, 'get_receipts_for_transactions_missing_them', side_effect=mock_decode_1),  # noqa: E501
+    ):
+        response = requests.post(
+            api_url_for(rotkehlchen_api_server, 'evmpendingtransactionsdecodingresource'),
+            json={'async_query': False, 'ignore_cache': True, 'chains': ['gnosis']},
+        )
+        assert_proper_response(response)
+
+    assert mock_monerium.call_args_list == [call(tx_hash=monerium_tx)]
+    # event update is only called once since the remote query function is mocked
+    assert mock_gnosis_pay_update_events.call_count == 1
+    assert mock_gnosis_pay_update_events.call_args_list[0].kwargs['transaction'].tx_hash == gnosis_pay_tx_1  # noqa: E501
+    # remote is only queried for the tx not in the database.
+    assert mock_gnosis_pay_query_remote.call_args_list == [call(tx_timestamp=ts_ms_to_sec(gnosis_pay_ts_2))]  # noqa: E501
+
+    # Also check that when only decoding new transactions, only the new events are updated.
+    new_event_tx_hash = make_evm_tx_hash()
+
+    def mock_decode_2(**kwargs: Any) -> list[EVMTxHash]:
+        """Add another gnosis pay event."""
```

Comment:
not a fun of functions inside the tests. Do you really need them? Can't you just add the entries and return the hashes that you need?